functorch/csrc/BatchRulesScatterOps.cpp

```diff
@@ -122,10 +161,10 @@ void index_put__batch_rule(
   if (!self_bdim.has_value()) {
     vmapIncompatibleInplaceError("index_put");
   }
-  auto self_ = moveBatchDimToFront(self, self_bdim);
-  auto values_ = moveBatchDimToFront(values, values_bdim);
-  TORCH_INTERNAL_ASSERT(indices.size() == indices_bdims.size());
-  std::vector<optional<Tensor>> indices_ = batchIndices(indices, indices_bdims, self_.size(0), self_bdim, values_bdim);
+  Tensor self_, values_;
+  std::vector<optional<Tensor>> indices_;
+  std::tie(self_, indices_, values_) = index_put_batch_rule_helper(
```

Comment:
`batchIndices` is used in the implementation of `index_batch_rule`. Is that one OK?

Reply:
Nvm I see what is going on now